src/App.ts

```diff
@@ -765,11 +765,10 @@ export default class App {
 
     // Set respond() utility
     if (body.response_url) {
-      listenerArgs.respond = (response: string | RespondArguments): Promise<any> => {
-        const validResponse: RespondArguments = typeof response === 'string' ? { text: response } : response;
-
-        return this.axios.post(body.response_url, validResponse);
-      };
+      listenerArgs.respond = buildRespondFn(this.axios, body.response_url);
+    } else if (typeof body.response_urls !== 'undefined' && body.response_urls.length > 0) {
+      // This can exist only when view_submission payloads - response_url_enabled: true
+      listenerArgs.respond = buildRespondFn(this.axios, body.response_urls[0].response_url);
```

Comment:
Just for my own knowledge, are we only handling the first `response_urls[0]` because that keeps the functionality consistent with other areas of Bolt-JS that only handle the first response url?

Reply:
@mwbrooks Good point. As of today, the `response_urls` can have only one element and we don't have any plans to have more elements in the short term. That's the reason why we can safely pick up the first element here, and thus far we can provide `respond` utility, which is consistent with other areas.

If the array start having multiple options in the future, we still can provide `respond` with the first element. When a developer would like to use a non-primary element, the right way to do so is to select the one from the array and then use axios directly with the URL. We may want to provide an easy-to-use utility for it.